packages/apollo-server/src/utils/types.ts

```diff
@@ -0,0 +1,90 @@
+import { GraphQLSchema } from 'graphql';
+import { CorsOptions } from 'cors';
+import { SchemaDirectiveVisitor, IResolvers } from 'graphql-tools';
+import { ConnectionContext } from 'subscriptions-transport-ws';
+import { GraphQLOptions } from 'apollo-server-core';
+
+export type Context<T = any> = T;
+export type ContextFunction<T = any> = (
+  context: Context<T>,
+) => Promise<Context<T>>;
+
+export interface SubscriptionServerOptions {
+  path?: string;
+  onConnect?: Function;
+  onDisconnect?: Function;
+  keepAlive?: number;
+}
+
+export interface Config<Server, ContextShape = any, Cors = CorsOptions>
+  extends Pick<
+      GraphQLOptions<Context<ContextShape>>,
+      | 'formatError'
+      | 'debug'
+      | 'rootValue'
+      | 'logFunction'
+      | 'formatParams'
+      | 'validationRules'
+      | 'formatResponse'
+      | 'fieldResolver'
+      | 'debug'
+      | 'cacheControl'
+      | 'tracing'
+    > {
+  app?: Server;
+  typeDefs?: string | [string];
+  resolvers?: IResolvers;
+  schema?: GraphQLSchema;
+  schemaDirectives: Record<string, typeof SchemaDirectiveVisitor>;
+  context?: Context<ContextShape> | ContextFunction<ContextShape>;
+  cors?: Cors;
+  subscriptions?: SubscriptionServerOptions | string | false;
+  engineInRequestPath?: boolean;
+  engine?: boolean | Object;
+  introspection?: boolean;
+}
+
+export interface EngineConfig {
```

Comment:
Can we get this from apollo-engine? Will that work if it's a devDependency?

Also this isn't engine *config*. This is engine LauncherOptions. Totally different thing.

Reply:
we can't since its not exported currently (we could change that) [here](https://github.com/apollographql/apollo-engine-js/blob/master/src/index.ts#L1-L10)

I'll change the name

Reply:
right, would be fine to export it

Reply:
I think we should do this prior to finalizing 2.0.